packages/core-app-api/src/apis/implementations/AppTranslationApi/AppTranslation.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AppTranslationApi,
+  LocalConfig,
+  Locals,
+} from '@backstage/core-plugin-api';
+import { Observable } from '@backstage/types';
+import { BehaviorSubject } from '../../../lib/subjects';
+import i18n, { i18n as i18nType } from 'i18next';
+import { initReactI18next } from 'react-i18next';
+
+const STORAGE_KEY = 'language';
+const DEFAULT_OPTIONS = {
+  lng: 'en',
+  interpolation: {
+    escapeValue: false,
+  },
+  languageOptions: [],
+};
+
+/**
+ * Exposes the locals installed in the app, and permits switching the currently
+ * active language.
+ *
+ * @public
+ */
+export class AppTranslation implements AppTranslationApi {
+  instance!: i18nType;
+
+  static createWithStorage(options?: LocalConfig) {
+    const finallyOptions = {
+      ...DEFAULT_OPTIONS,
+      ...(options || {}),
+    };
+
+    if (finallyOptions?.resources) {
+      const languages = Object.keys(finallyOptions.resources);
+      finallyOptions.languageOptions = new Set([
+        ...finallyOptions.languageOptions,
+        ...languages,
+      ]);
+    }
+
+    const i18nInstance = i18n.createInstance();
+    i18nInstance.use(initReactI18next).init(finallyOptions);
```

Comment:
Why not just use the i18next plugins for `lazyResources` and `storage` like [`i18next-http-backend`](https://github.com/i18next/i18next-http-backend) and [`i18next-browser-languageDetector`](https://github.com/i18next/i18next-browser-languageDetector)?